crates/oxc_linter/src/rules/no_dupe_keys.rs

```diff
@@ -0,0 +1,110 @@
+use oxc_ast::{
+    ast::{ObjectProperty, PropertyKind},
+    AstKind, GetSpan, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use rustc_hash::FxHashMap;
+
+use crate::{ast_util::calculate_hash, context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-dupe_keys): Disallow duplicate keys in object literals")]
+#[diagnostic(severity(warning), help("Consider removing the duplicated key"))]
+struct NoDupeKeysDiagnostic(#[label] pub Span, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoDupeKeys;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow duplicate keys in object literals
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Multiple properties with the same key in object literals can cause unexpected behavior in your application.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var foo = {
+    ///     bar: "baz",
+    ///     bar: "qux"
+    /// }
+    /// ```
+    NoDupeKeys,
+    correctness
+);
+
+impl Rule for NoDupeKeys {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::ObjectExpression(obj_expr) = node.get().kind() {
+            let mut map = FxHashMap::default();
+            for prop in obj_expr.properties.iter() {
+                if let ObjectProperty::Property(prop) = prop && let Some(key_name) = prop.key.static_name().as_ref() {
+                    let hash = calculate_hash(key_name);
+                    if map.contains_key(&hash) {
+                        let (kind, span) = map.get(&hash).unwrap();
+                        if kind == &PropertyKind::Init {
+                            ctx.diagnostic(NoDupeKeysDiagnostic(*span, prop.key.span()));
+                        } else if prop.kind == PropertyKind::Init {
+                            ctx.diagnostic(NoDupeKeysDiagnostic(*span, prop.key.span()));
+                            map.insert(hash, (prop.kind, prop.key.span()));
```

Comment:
This seems redundant, just let it report the newest two duplications is fine I think.

Reply:
I think we need to check whether the the kind is `Init` or not. Please consider this case: `foo{a: 1, get a(){}, set a(value){}}`.
If we directly insert the tuple, we cannot point out that `a` and `set a` are duplicated, we can only point out that `a` and `get a` are duplicated.